src/lib.rs

```diff
@@ -1608,6 +1608,53 @@ pub trait Itertools : Iterator {
         None
     }
 
+    /// Check whether the iterator contains an item.
+    ///
+    /// If the iterator contains the item prior to its end,
+    /// this method will short-circuit and only partially
+    /// exhaust the iterator.
+    ///
+    /// ```
+    /// use itertools::Itertools;
+    ///
+    /// let data = vec![
+    ///      "foo".to_owned(),
+    ///      "bar".to_owned(),
+    ///      "baz".to_owned(),
+    /// ];
+    /// // this could get expensive:
+    /// assert!(data.contains(&"foo".to_owned()));
+    /// // this, less so:
+    /// assert!(data.iter().contains("foo"));
```

Comment:
What makes the former example expensive, and the latter example cheap?

Reply:
The former example copies the contents of the given &str from wherever it was borrowed into a new heap allocation, and then searches the Vec for it; the latter example does the search without doing the copy-of-contents.